src/image_bios.cpp

```diff
@@ -378,16 +382,52 @@ void BIOSUpdater::doBeforeInstall(bool reset)
 
     if (!reset)
     {
-        puts("Preserving NVRAM...");
-        const fs::path dumpFile(tmpdir / nvramFile);
-        const std::string cmd = strfmt(
-            "dd if=%s of=%s skip=%lu count=%lu", mtdDevice, dumpFile.c_str(),
-            nvramOffset / ddBlockSize, nvramSize / ddBlockSize);
-        const int rc = system(cmd.c_str());
-        checkWaitStatus(rc, std::string());
-        if (!fs::exists(dumpFile))
+        // Migrating UEFI settings from previous version (1.3 or earlier)
+        puts("Import UEFI settings...");
+        bool isSettingsExist = false;
+        try
         {
-            throw FwupdateError("Error reading NVRAM");
+            // Check if uefivar service still doesn't have settings
+            auto method =
+                systemBus.new_method_call(uefivarInterface, uefivarObject,
+                                          uefivarInterface, "NextVariable");
+            method.append(std::string(), std::vector<uint8_t>(16, 0));
+            systemBus.call_noreply(method);
+            isSettingsExist = true;
+        }
+        catch (const sdbusplus::exception::SdBusError& ex)
+        {
+            if (strcmp(ex.name(),
+                       "xyz.openbmc_project.Common.Error.NotAllowed") != 0)
+            {
+                throw FwupdateError("Unable to import NVRAM: %s", ex.what());
```

Comment:
`SdBusError::what()` contains the original exception class name and so on.
Likely `SdBusError::description()` would be more appropriate here.

Reply:
I don't get it.
Do you want to replace `ex.what()` with `ex.description()`?

Reply:
yes

Reply:
Fixed